coffee/shepherd.coffee

```diff
@@ -117,6 +117,9 @@ class Step extends Evented
     @tether = new Tether extend(tetherOpts, @options.tetherOptions)
 
   show: =>
+      @options.beforeShowPromise?()?.then(=> @_show()) ? @_show()
```

Comment:
Using the fat arrow (`=>`) to define `_show` adds a bind function call in the constructor to the instance when compiled. This makes it so that any function call to `_show` will in fact be the instance of the `Step` it was invoked in.